### PR TITLE
Weigh gilknocker Prometheus metric by duration

### LINKED ISSUE #8557: GIL prometheus metrics are misleading
I've run a benchmark on a 2-workers cluster that simulates these use cases:
1. a C extension repeatedly blocks the GIL on one worker in 1 second blocks
2. a C extension repeatedly blocks the GIL on one worker  in 4 second blocks
3. a C extension repeatedly blocks the GIL on one worker  in 15 second blocks
4. a hot pure-Python loop repeatedly blocks the GIL on one worker  in 15 seconds blocks

In all cases, the Bokeh dashboard displays the GIL metric more or less constantly at 50%. This is improvable, as it hides the disparity between the workers (it would be better IMHO to have something like min+median+max), but otherwise it is accurate.

The same metric on Prometheus (which, on Coiled, scrapes every 5 seconds), however, is misleading.
In all cases, **I would expect the metric for the affected worker to read 100%**. Instead it reads:
1. 72%
2. 20~25%
3. ~7% (failed to scrape most times)
4. 162%

In the first three cases, this leads to a misleading sense that the GIL is not being problematic.
The last case makes no sense to me - how can GIL contention be more than 100%? (again, on the bokeh dashboard it shows 50% cluster average, or 100% on the affected worker).

image

This uses coiled but a plain LocalCluster will yield the same (assuming you have prometheus scraping it).
## Reproducer
```python
import coiled
import dask
import distributed
import ctypes
from time import time, sleep

cluster = coiled.Cluster(n_workers=2)
client = distributed.Client(cluster)
a, b = client.has_what()

def extension_hogs_gil(seconds):
    libc = ctypes.PyDLL('libc.so.6')
    libc.sleep(seconds)


def pure_pyton_hogs_gil(seconds):
    deadline = time() + seconds
    fib0, fib1 = 0, 1
    n = 2
    while True:
        if n % 10_000 == 0 and time() > deadline:
            return n
        fib0, fib1 = fib1, fib0 + fib1
        n += 1


for _ in range(120):
    client.submit(extension_hogs_gil, 1, pure=False, workers=[a]).result()
sleep(10)

for _ in range(120//4):
    client.submit(extension_hogs_gil, 4, pure=False, workers=[a]).result()
sleep(10)

for _ in range(120//15):
    client.submit(extension_hogs_gil, 15, pure=False, workers=[a]).result()
sleep(10)

for _ in range(120//15):
    client.submit(pure_pyton_hogs_gil, 15, pure=False, workers=[a]).result()
```

### PR DESCRIPTION
**Pull request by crusaderky**
- Blocked by #8560
- Closes #8557
- Renames prometheus metrics:

| main | this PR |
| --- | --- |
| dask_scheduler_gil_contention_total | dask_scheduler_gil_contention_seconds_total |
| dask_worker_gil_contention_total | dask_worker_gil_contention_seconds_total |

stress test from #8557:
## BEFORE
Screenshot from 2024-03-06 15-17-22

## AFTER
Screenshot from 2024-03-07 16-25-14

CC @ntabris @milesgranger

### CODE REVIEW

File: distributed/system_monitor.py [hunk 1]
```diff
@@ -114,12 +114,11 @@ def __init__(
...
             else:
                 self.quantities["gil_contention"] = deque(maxlen=maxlen)
-                self._cumulative_gil_contention = 0.0
+                self.cumulative_gil_contention = 0.0
                 raw_interval = dask.config.get(
                     "distributed.admin.system-monitor.gil.interval",
                 )
-                interval = parse_timedelta(raw_interval, default="us") * 1e6
-
+                interval = parse_timedelta(raw_interval) * 1e6
```
Comment (crusaderky) [inline distributed/system_monitor.py:121]: Don't interpret a raw `0.001` in the config (which violates the jsonschema) as 1us. Crash instead.

File: distributed/system_monitor.py [hunk 2]
```diff
@@ -197,10 +196,10 @@ def update(self) -> dict[str, Any]:
             self._last_host_cpu_counters = host_cpu
 
         if self.monitor_gil_contention:
-            self._last_gil_contention = self._gilknocker.contention_metric
-            self._cumulative_gil_contention += self._last_gil_contention
-            result["gil_contention"] = self._last_gil_contention
+            gil_contention = self._gilknocker.contention_metric
             self._gilknocker.reset_contention_metric()
+            result["gil_contention"] = self._last_gil_contention = gil_contention
+            self.cumulative_gil_contention += duration * gil_contention
```
Comment (crusaderky) [inline distributed/system_monitor.py:202]: This is the key change.

File: docs/source/prometheus.rst [hunk 3]
```diff
@@ -26,9 +26,11 @@ dask_scheduler_clients
     Number of clients connected
 dask_scheduler_desired_workers
     Number of workers scheduler needs for task graph
-dask_scheduler_gil_contention_total
-    Value representing cumulative total of GIL contention,
-    in the form of summed percentages.
+dask_scheduler_gil_contention_seconds_total
+    Value representing cumulative total of *potential* GIL contention,
+    in the form of cumulative seconds during which a thread held the GIL locked.
```
Comment (ntabris) [inline docs/source/prometheus.rst:31]:
```suggestion
    in the form of cumulative seconds during which any thread held the GIL locked.
```

nit but this feels slightly clearer to me ("a thread" initially made me think it was for some specific individual thread, and then I was confused)

**Comment by ntabris** [comment]
Thanks, @crusaderky! It would be nice to be able to get more accurate GIL data via Prometheus.

**Review by hendrikmakait — APPROVED** [review]
Thanks, @crusaderky! Changes look good, assuming CI turns green.

**Review by milesgranger — APPROVED** [review]
Yep, good for me as well, sorry for the delayed review here!

**Pull request merged**